csharp/device/Microsoft.Azure.Devices.Client/Common/Security/SharedAccessSignatureBuilder.cs

```diff
@@ -104,6 +151,34 @@ static string Sign(string requestString, string key)
             var sign = CryptographicBuffer.EncodeToBase64String(hash.GetValueAndReset());
             return sign;
         }
+#elif NETMF
+        static string Sign(string requestString, string key)
+        {
+            // computing SHA256 signature using a managed code library
+            var hmac = SHA.computeHMAC_SHA256(Convert.FromBase64String(key), Encoding.UTF8.GetBytes(requestString));
+            return Convert.ToBase64String(hmac);
+
+            // computing SHA256 signature using the .NET Micro Framework classes
```

Comment:
as I understand it it's useful to keep this commented out in case someone wants to switch to another set of (presumably hardware-based) crypto API